### PR TITLE
Serverless test against restricted API key

### PR DESCRIPTION
**Pull request by kaisecheng**
This commit changes the serverless tests to use a restricted API key with necessary permissions instead of user/ password of super user. Also, it deletes Logstash legcy monitoring test and fixes Kibana tests that compare strings with curly brackets.

AWS Serverless endpoints and credentials are updated in Vault. Two more keys are added. `LS_ROLE_API_KEY_ENCODED` is base64 of api_key and `LS_PLUGIN_API_KEY` is in the format id:api_key that requires by Logstash.

API key for Logstash is generated with the following privileges.
```
      "cluster": ["monitor", "manage_index_templates", "manage_logstash_pipelines", "cluster:admin/ingest/pipeline/get", "read_pipeline"], 
      "indices": [
        {
          "names": [ "logstash", "logstash-*", "ecs-logstash", "ecs-logstash-*", "serverless*", "logs-*", "metrics-*", "synthetics-*", "traces-*" ], 
          "privileges": ["manage", "write", "create_index", "read", "view_index_metadata"]  
        }
      ]
```

### CODE REVIEW

**Comment by kaisecheng** [comment]
sample [build](https://buildkite.com/elastic/logstash-serverless-integration-testing/builds/75)

File: .buildkite/serverless_integration_pipeline.yml [hunk 1]
```diff
@@ -16,8 +16,9 @@ steps:
     command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/elastic_integration_filter_tests.sh
   - label: "central pipeline management test"
     command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/cpm_tests.sh
-  - label: "Logstash legacy monitoring test"
-    command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/monitoring_tests.sh
+# Legacy monitoring is disabled. Serverless does not support /_monitoring/bulk, hence the test always fails to ingest metrics.
+#  - label: "Logstash legacy monitoring test"
+#    command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/monitoring_tests.sh
```
Comment (jsvd) [inline .buildkite/serverless_integration_pipeline.yml:21]: Looking at https://github.com/buildkite/feedback/issues/55#issuecomment-305028566 we should be able to just skip it:

```suggestion
# Legacy monitoring is disabled. Serverless does not support /_monitoring/bulk, hence the test always fails to ingest metrics.
- label: "Logstash legacy monitoring test"
  command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/monitoring_tests.sh
  skip: true
```

**Review by jsvd — COMMENTED** [review]
Overall LGTM, just suggested using a skip primitive in buildkite instead of commenting out the code.
Also, I think we should start collecting documentation on the setup procedures for these tests in the repo instead of github comments. For example, I suggest adding a README.md to the `ci/serverless` dir that explains how to create the user and api keys needed for the tests.

File: .buildkite/serverless_integration_pipeline.yml [hunk 2]
```diff
@@ -16,8 +16,10 @@ steps:
     command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/elastic_integration_filter_tests.sh
   - label: "central pipeline management test"
     command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/cpm_tests.sh
-  - label: "Logstash legacy monitoring test"
-    command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/monitoring_tests.sh
+# Legacy monitoring is disabled. Serverless does not support /_monitoring/bulk, hence the test always fails to ingest metrics.
+- label: "Logstash legacy monitoring test"
+  command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/monitoring_tests.sh
+  skip: true
```
Comment (jsvd) [inline .buildkite/serverless_integration_pipeline.yml:22]: fixing indentation:

```suggestion
  # Legacy monitoring is disabled. Serverless does not support /_monitoring/bulk, hence the test always fails to ingest metrics.
  - label: "Logstash legacy monitoring test"
    command: ./.buildkite/scripts/setup_java.sh && ./ci/serverless/monitoring_tests.sh
    skip: true
```

**Review by jsvd — APPROVED** [review]
LGTM, minor fix in indentation

**Pull request merged**